### DISCUSSION TITLE
I need a pro in this!

### DISCUSSION THREAD

**Discussion by DJRamsadOriginal** [bmaltais/kohya_ss#2302] [Q&A]
I wanted to create my own LoRa but there seems to be a problem with a file....I did not change any file locations....

Error-Code:
[Errno 2] No such file or directory
Traceback (most recent call last):
  File "F:\python\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "F:\python\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\ai lora ka\kohya_ss-master\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 1017, in launch_command
    simple_launcher(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 637, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['E:\\ai lora ka\\kohya_ss-master\\venv\\Scripts\\python.exe', 'E:\\ai lora ka\\kohya_ss-master/sd-scripts/train_db.py', '--bucket_no_upscale', '--bucket_reso_steps=64', '--cache_latents', '--caption_extension=.txt', '--enable_bucket', '--min_bucket_reso=256', '--max_bucket_reso=2048', '--huber_c=0.1', '--huber_schedule=snr', '--learning_rate=1e-05', '--learning_rate_te=1e-05', '--logging_dir=E:/ai lora ka/kohya_ss-master/outputs\\log', '--loss_type=l2', '--lr_scheduler=cosine', '--lr_scheduler_num_cycles=1', '--lr_warmup_steps=160', '--max_data_loader_n_workers=0', '--resolution=512,512', '--max_train_steps=1600', '--min_timestep=0', '--mixed_precision=fp16', '--optimizer_type=AdamW8bit', '--output_dir=E:/ai lora ka/kohya_ss-master/outputs\\model', '--output_name=last', '--pretrained_model_name_or_path=stabilityai/stable-diffusion-2-1-base', '--save_every_n_epochs=1', '--save_model_as=safetensors', '--save_precision=fp16', '--train_batch_size=1', '--train_data_dir=E:/ai lora ka/kohya_ss-master/outputs\\img', '--v2', '--xformers']' returned non-zero exit status 2.

**Comment by 5KilosOfCheese** [comment]
You have accidentally selected the Dreambooth tab and attempted to train Dreambooth. In the GUI select LoRA and try again.
![kuva](https://github.com/bmaltais/kohya_ss/assets/117771081/ba019995-2d3f-447b-a4b7-b677f0021b3a)

**Reply by DJRamsadOriginal** [reply]
Well, thank you for your help. Sadly it didn't change anything. I am still getting kind of the same error...I really hope that you could help me with that issue. For me it ssems like my Python has problems. I already re-installed it, as same as kohya_ss.

F:\python\python.exe: can't open file 'E:\\ai lora ka\\kohya_ss-master\\sd-scripts\\sdxl_train_network.py': [Errno 2] No such file or directory
Traceback (most recent call last):
  File "F:\python\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "F:\python\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\ai lora ka\kohya_ss-master\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 1017, in launch_command
    simple_launcher(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 637, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['E:\\ai lora ka\\kohya_ss-master\\venv\\Scripts\\python.exe', 'E:\\ai lora ka\\kohya_ss-master/sd-scripts/sdxl_train_network.py', '--bucket_no_upscale', '--bucket_reso_steps=64', '--cache_latents', '--cache_latents_to_disk', '--caption_extension=.txt', '--enable_bucket', '--min_bucket_reso=256', '--max_bucket_reso=2048', '--gradient_accumulation_steps=2', '--gradient_checkpointing', '--huber_c=0.1', '--huber_schedule=snr', '--learning_rate=0.0009', '--logging_dir=E:/ai lora ka/Output Leggings 1\\log', '--loss_type=l2', '--lr_scheduler=constant', '--lr_scheduler_num_cycles=1', '--max_data_loader_n_workers=0', '--max_grad_norm=1', '--resolution=1024,1024', '--max_train_steps=5143', '--min_timestep=0', '--mixed_precision=bf16', '--network_alpha=1', '--network_dim=128', '--network_module=networks.lora', '--no_half_vae', '--optimizer_args', 'scale_parameter=False', 'relative_step=False', 'warmup_init=False', '--optimizer_type=Adafactor', '--output_dir=E:/ai lora ka/Output Leggings 1\\model', '--output_name=last', '--pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5', '--save_every_n_epochs=1', '--save_model_as=safetensors', '--save_precision=bf16', '--seed=12345', '--text_encoder_lr=0.0009', '--train_batch_size=7', '--training_comment=trigger: supergirl costume', '--train_data_dir=E:/ai lora ka/Output Leggings 1\\img', '--unet_lr=0.0009', '--xformers']' returned non-zero exit status 2.

**Reply by 5KilosOfCheese** [reply]
Check whether #1744 applies, ensure that no folder or file part of the training process has "unusual letters". Because if you did a fresh install everything should be working correctly, assuming there has been no issues earlier; or is this your first time trying to train something?

**Reply by DJRamsadOriginal** [reply]
This is my first time. I did do training before but with civitai, I wanted to do it locally with my rtx. I can try to re-install it again.

**Reply by 5KilosOfCheese** [reply]
Did the fix I refrenced help at all? Deleting Venv and reinstalling is usually a good fix, however if the issue persists it is something you are doing. 

However do tell if the reinstall helps. 

If it doesn't then try redoing the folder structure and files names, just add one picture to the image folder and don't enable captions. Just to see whether the training starts at all.

**Reply by DJRamsadOriginal** [reply]
Hey ho...I did the reinstallation but it did not work. I treid it out with 1 picture but it still does not work. I have done it without captioning and with captioning. Maybe I am doing something wrong, I mean, I am pretty new to AI and LoRa and I try to understand. 

How ever, I tried to use 4 different ssd's as path but no change, still get an error.

The following values were not passed to `accelerate launch` and had defaults used instead:
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
F:\python\python.exe: can't open file 'E:\\ai lora ka\\kohya_ss-master\\sd-scripts\\train_network.py': [Errno 2] No such file or directory
Traceback (most recent call last):
  File "F:\python\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "F:\python\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\ai lora ka\kohya_ss-master\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 1017, in launch_command
    simple_launcher(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 637, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['E:\\ai lora ka\\kohya_ss-master\\venv\\Scripts\\python.exe', 'E:\\ai lora ka\\kohya_ss-master/sd-scripts/train_network.py', '--bucket_no_upscale', '--bucket_reso_steps=64', '--cache_latents', '--caption_extension=.txt', '--enable_bucket', '--min_bucket_reso=256', '--max_bucket_reso=2048', '--huber_c=0.1', '--huber_schedule=snr', '--learning_rate=0.0001', '--logging_dir=C:/Users/rasa1/OneDrive/Desktop/1\\log', '--loss_type=l2', '--lr_scheduler=cosine', '--lr_scheduler_num_cycles=1', '--lr_warmup_steps=8', '--max_data_loader_n_workers=0', '--max_grad_norm=1', '--resolution=512,512', '--max_train_steps=80', '--min_timestep=0', '--mixed_precision=fp16', '--network_alpha=1', '--network_dim=8', '--network_module=networks.lora', '--optimizer_type=AdamW8bit', '--output_dir=C:/Users/rasa1/OneDrive/Desktop/1\\model', '--output_name=last', '--pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5', '--save_every_n_epochs=1', '--save_model_as=safetensors', '--save_precision=fp16', '--text_encoder_lr=0.0001', '--train_batch_size=1', '--train_data_dir=C:/Users/rasa1/OneDrive/Desktop/1\\img', '--unet_lr=0.0001', '--xformers']' returned non-zero exit status 2.


I sadly can't tell what I am doing wrong...because I am following the tutorial steps that are shown at the guides bar.

**Reply by 5KilosOfCheese** [reply]
Ah! Ok! After the installation process did you configure accelerate? You can do it again with option 5 in the install. Just choose the defaults.

![kuva](https://github.com/bmaltais/kohya_ss/assets/117771081/8c31b0b0-45de-421a-9228-3f915b604ef4)

**Reply by DJRamsadOriginal** [reply]
Okay, well. It did change something. But I still dont get it because it says that it can't find a directory. Pretty weird if you ask me.

14:57:50-468118 INFO     accelerate launch --mixed_precision="fp16" --num_processes=1 --num_machines=1
                         --num_cpu_threads_per_process=2 "E:\ai lora ka\kohya_ss-master/sd-scripts/train_network.py"
                         --bucket_no_upscale --bucket_reso_steps=64 --cache_latents --caption_extension=".txt"
                         --enable_bucket --min_bucket_reso=256 --max_bucket_reso=2048 --huber_c="0.1"
                         --huber_schedule="snr" --learning_rate="0.0001" --logging_dir="E:/ai lora ka/Output Leggings
                         1\log" --loss_type="l2" --lr_scheduler="cosine" --lr_scheduler_num_cycles="1"
                         --lr_warmup_steps="320" --max_data_loader_n_workers="0" --max_grad_norm="1"
                         --resolution="512,512" --max_train_steps="3200" --min_timestep=0 --mixed_precision="fp16"
                         --network_alpha="1" --network_dim=8 --network_module=networks.lora --optimizer_type="AdamW8bit"
                         --output_dir="E:/ai lora ka/Output Leggings 1\model" --output_name="last"
                         --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" --save_every_n_epochs="1"
                         --save_model_as=safetensors --save_precision="fp16" --text_encoder_lr=0.0001
                         --train_batch_size="1" --train_data_dir="E:/ai lora ka/Output Leggings 1\img" --unet_lr=0.0001
                         --xformers
[2024-04-19 14:57:56,487] torch.distributed.elastic.multiprocessing.redirects: [WARNING] NOTE: Redirects are currently not supported in Windows or MacOs.
14:57:57-136410 INFO     Copy C:/Users/rasa1/OneDrive/Desktop/Leggings to E:/ai lora ka/Output Leggings
                         1\img/40_leggings clothing...
[W socket.cpp:663] [c10d] The client socket has failed to connect to [DESKTOP-DJBCVTL]:29500 (system error: 10049 - Die angeforderte Adresse ist in diesem Kontext ung³ltig.).
F:\python\python.exe: can't open file 'E:\\ai lora ka\\kohya_ss-master\\sd-scripts\\train_network.py': [Errno 2] No such file or directory
[2024-04-19 14:58:04,672] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 2) local_rank: 0 (pid: 27480) of binary: E:\ai lora ka\kohya_ss-master\venv\Scripts\python.exe
Traceback (most recent call last):
  File "F:\python\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "F:\python\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\ai lora ka\kohya_ss-master\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 1008, in launch_command
    multi_gpu_launcher(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\accelerate\commands\launch.py", line 666, in multi_gpu_launcher
    distrib_run.run(args)
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\torch\distributed\run.py", line 797, in run
    elastic_launch(
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\torch\distributed\launcher\api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "E:\ai lora ka\kohya_ss-master\venv\lib\site-packages\torch\distributed\launcher\api.py", line 264, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:
============================================================
E:\ai lora ka\kohya_ss-master/sd-scripts/train_network.py FAILED
------------------------------------------------------------
Failures:
  <NO_OTHER_FAILURES>
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2024-04-19_14:58:04
  host      : DESKTOP-DJBCVTL
  rank      : 0 (local_rank: 0)
  exitcode  : 2 (pid: 27480)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================

**Reply by 5KilosOfCheese** [reply]
OK. Are you trying to run this whole thing from OneDrive or otherwise remotely? Download the model on to your computer, along with kohya and the datafolders for training, and see if that helps. Because there are errors here which don't actually refer to anything that is really within the realm of this GUI or Kohya as far as I can tell.

**Reply by DJRamsadOriginal** [reply]
I am running it complete locally. I treid it out, did not change anything, sadly. Same issue as above.

**Reply by wisamidris7** [reply]
Change your computer.

**Comment by b-fission** [comment]
Super late reply, but the problem is in this log line.
`F:\python\python.exe: can't open file 'E:\ai lora ka\kohya_ss-master\sd-scripts\sdxl_train_network.py': [Errno 2] No such file or directory`

If you downloaded kohya_ss as a zip file from github, it's best **not** to do that. You're expected to use git to clone the repo. 

But if you really don't care about git, then just grab [sd-scripts from here](https://github.com/kohya-ss/sd-scripts) and extract the contents to a folder called `sd-scripts` so that the path to kohya_ss-master\sd-scripts\sdxl_train_network.py is valid.

....

But then your next set of errors for `torch.distributed.elastic.multiprocessing` would indicate you've configured `accelerate` to use distributed training, which... if you have to ask, you probably don't need it. So go to the setup utility, choose option 5 (Manually configure accelerate), and press Enter for all the defaults.
